### ISSUE TITLE
Xamarin app: all works but after a few minutes... no more messages received... 

### ISSUE THREAD

**Issue by AngeloLazzari** [chkr1011/MQTTnet#948]
Hi, we created a client app that connects to the brocker with 

```csharp
var options = new MqttClientOptionsBuilder()
                .WithClientId(userId)
                .WithTcpServer(mqttBrokerHost, mqttBrokerPort)
                .WithCredentials(mqttBrokerUsername, mqttBrokerPassword)
                // .WithCommunicationTimeout(TimeSpan.FromDays(1))
                .Build();

            try
            {

                // CHECHINK IF THE CLIENT IS ALREADY CONNECTED
                // AND IF SO, CLOSE CONNECTION

                if (! mqttClient.IsConnected)
                {
                    // CONNECT TO BROKER
                    await mqttClient.ConnectAsync(options, CancellationToken.None);
                }
                
            }
            catch (Exception ex)
            {
                string error = ex.Message;
                
            }
```

than we do a 
`            await mqttClient.SubscribeAsync(new MqttTopicFilter { Topic = subscribeTopicReq });
`
we tried to use the ConnectedHandler tu do the SubscribeAsync but it seems is not fired.... so we are using the direct SubscribeAsync

for a while all is working ... all messages are coming to the client ...but, after a while the app stop receiving the messages...

we implemented the mqttClient.UseDisconnectedHandler and we even call the reconnection if internet connection change from 4g to wifi and viceversa.... 

It seems that the UseDisconnectedHandler is not always fired .... 

do you have any suggestion? 
Do we need to develop in somehow a background service that check every any time if the connection is active?

We noticed that we cannot call mqttClient.IsConnected method... how can we check that info? 

thank you very much
Angelo

**Comment by SeppPenner** [comment]
@AngeloLazzari Can you maybe tell me what the issue was and how you resolved it?

**Comment by AngeloLazzari** [comment]
Hi @SeppPenner thanks for your reply. I closed the issue because I opened a new one #951. We discovered that, until the mobile device is plugged into the power source, all is working correctly...but, when you unplug it, after a few seconds, the library stops to send keepalive push to the broker...and the broker disconnects the client. Right now we put in stand by the proyjct because we need to solve this problem....
